concept/thing/thing_manager.rs

```diff
@@ -386,61 +385,90 @@ impl ThingManager {
             return Ok(AttributeIterator::new_empty());
         };
 
-        let Some((range_start, range_end)) = Self::get_value_range(attribute_value_type, range, |value| {
-            AttributeVertex::build_prefix_for_value(
-                attribute_type.vertex().type_id_(),
-                value,
-                self.vertex_generator.hasher(),
-            )
-        })?
-        else {
+        let Some((value_lower_bound, value_upper_bound)) = Self::get_value_range(attribute_value_type, range)? else {
             return Ok(AttributeIterator::new_empty());
         };
-        let key_range_start = match range_start {
-            Bound::Included(start) => RangeStart::Inclusive(start),
-            Bound::Excluded(start) => RangeStart::Exclusive(start),
+
+        let start_attribute_vertex_prefix_range = match value_lower_bound {
```

Comment:
We now take a `impl RangeBounds`, since the API is more restrictive than our `KeyRange` type. Each method here translates the RangeBounds into a slightly different KeyRange, but we consistently extend the range to be as long as possible (including using the type ID `0` when no target type is provided), which should generally hit a longer prefix.

I'm actually not entirely sure, thinking more about how RocksDB uses prefix filters, that this is actually useful. But at least now we're using a consistent prefix length of searches...